sql-plugin/src/main/spark332db/scala/org/apache/spark/sql/rapids/shims/GpuDataSource.scala

```diff
@@ -83,9 +84,7 @@ case class GpuDataSource(
     PartitioningUtils.validatePartitionColumn(data.schema, partitionColumns, caseSensitive)
 
     val fileIndex = catalogTable.map(_.identifier).map { tableIdent =>
-      sparkSession.table(tableIdent).queryExecution.analyzed.collect {
-        case LogicalRelation(t: HadoopFsRelation, _, _, _) => t.location
-      }.head
+      LogicalPlanShims.getLocations(sparkSession.table(tableIdent).queryExecution.analyzed).head
```

Comment:
It is not a nice Scala but I recommend to future-proof and avoid small shim burden:
```Scala
    val fileIndex = catalogTable.map(_.identifier).map { tableIdent =>
      sparkSession.table(tableIdent).queryExecution.analyzed.collect {
        case lr: LogicalRelation if lr.relation.isInstanceOf[HadoopFsRelation] => 
          lr.relation.asInstanceOf[HadoopFsRelation].location
      }.head
    }
```

Reply:
I considered this at first, but I'm not sure this is "future-proof."  I think we'd rather prefer to have the compiler and/or runtime inform us if a Spark platform changed this interface, since we may need to do more than just ignore the added parameters.  In those cases Spark would have to change this corresponding code, and we would need to change it in a compatible way.  Maybe that change is to ignore the added or removed parameters, maybe not.